rpc/utils.go

```diff
@@ -250,5 +251,13 @@ func NewID() ID {
 			val >>= 8
 		}
 	}
-	return ID("0x" + hex.EncodeToString(id))
+
+	rpcId := hex.EncodeToString(id)
+	// rpc ID's are RPC quantities, no leading zero's and 0 is 0x0
+	rpcId = strings.TrimLeft(rpcId, "0")
+	if rpcId == "" {
+		rpcId = "0"
+	}
+
+	return ID("0x" + rpcId)
```

Comment:
Wouldn't it be easier/cleaner to

 * Extend `HexNumber` to have a `String` method (basically split the MarshalJSON into two funcs)
 * `return ID(NewHexNumber(new(big.Int).SetBytes(id)).String())`

Not the best looking, but would prevent duplicating non-obvious logic.

---

Another alternative (maybe the cleanest) would be `return ID(fmt.Sprintf("0x%x", new(big.Int).SetBytes(id)))`